api/v1alpha1/clusterextension_types.go

```diff
@@ -95,13 +105,18 @@ const (
 	ReasonResolutionUnknown         = "ResolutionUnknown"
 	ReasonSuccess                   = "Success"
 	ReasonDeprecated                = "Deprecated"
+	ReasonErrorGettingReleaseState  = "ErrorGettingReleaseState"
+	ReasonUpgradeFailed             = "UpgradeFailed"
+	ReasonCreateDynamicWatchFailed  = "CreateDynamicWatchFailed"
```

Comment:
Do we need reasons like `ErrorGettingClient`, `CreateDynamicWatchFailed`, etc? Do clients care about this?

Reply:
I find the matrix of possibilities a bit overwhelming. And I think we should embed the matrix in the repo docs. But on the other hand, the many reasons basically represent different things to log about which overall is good for debugging and getting better field bug reports.

Reply:
Logging and Condition reasons are different concerns. I agree with @m1kola here. Condition types are reasons are part of our public API. We should keep those as minimal as we possibly can.

Reasons are generally for software clients.
Messages are for humans. If there isn't a use case in mind for some other software needing to distinguish between reasons of a condition, we should not have explicit reasons for them. Just put the details in a message for a human to assess.

Reply:
There could be argument both ways, as I see clients wanting to know why a bundle which was supposed to be watched not triggering a reconcile, or why an installed version is unavailable. These were taken in from Rukpak. 

But for now, have removed it from operator-controller, and using the ones in Rukpak (which we don't intend to support as an API).